### DISCUSSION TITLE
BinSPreader read acquisition question

### DISCUSSION THREAD

**Discussion by aababc1** [ablab/spades#976] [Q&A] [answered]
Thank you very much for your hard and nice work for BinSpreader. 

I used BinSPreader to improve bins got using METABAT2. ( The raw metagenome paried fastq was quality-controlled. I used metaSPAdes to get scaffolds.fa and gfa file ) .  

I got binning.tsv table. It seems to be that they only get header information like below example.  

$head binning.tsv
NODE_1_length_606266_cov_11.265954	bin.9.fa
NODE_2_length_403483_cov_13.852217	bin.32.fa
NODE_3_length_375106_cov_96.289678	bin.17.fa
NODE_4_length_326388_cov_38.813120	bin.71.fa
NODE_5_length_325088_cov_8.357164	bin.94.fa
NODE_6_length_309800_cov_102.246470	bin.17.fa

It there any suggested way to get improved bins using BinSPreader using binining.tsv results? 

Thank you very much.

**Comment by asl** [comment, marked as answer]
Sure. There are lots of scripts around there. E.g. https://github.com/cmks/DAS_Tool/blob/master/src/Contigs2Bin_to_Fasta.sh

**Reply by aababc1** [reply]
Thank you so much for your quick reply.
It worked perfectly well.

**Comment by ebi-jlu8** [comment]
I have a followup question. I am using binspreader to improve bins I got from metawrap. I have used the Contigs2Bin_to_Fasta.sh to extract sequences from the refined binning.tsv file. I have a number of sequences which are not in the assembly.fasta but present in the assembly_scaffolds file. Is there a way to extract the matching sequence from the assembly graph (*.gfa) instead of the fasta file?

**Reply by asl** [reply]
Sorry, what is "assembly_scaffolds" file? This is not something produced by BinSPreader.

**Reply by ebi-jlu8** [reply]
Hi Anton. It's the assembly.gfa file that was produced by metaspades. I can see the header in the gfa file, but have no idea how to extract it.

**Reply by asl** [reply]
I do not know how `assembly.fa` corresponds to `assembly.gfa` in case of metawrap as they might post-process / alter the files produced by assembler. But normally (in case of SPAdes) scaffolds are represented as paths (`P` records) in assembly graph. So one would need to walk the segments, get the corresponding sequences, etc. It seems that MetaWRAP discards short scaffolds, I'd suggest you to take the complete file from assembler's output.

**Reply by ebi-jlu8** [reply]
Sorry. I meant to say metaspades. We used metaspades to get the output assembly_graph_with_scaffolds.gfa, contigs.fasta and scaffolds.fasta. I originally tried to recover the refined bins from the contigs.fasta file (after bin-refinement), but was missing a lot of sequences. I am not familiar with gfa files. Should I be looking in the `assembly_graph_with_scaffolds.gfa` file for binned sequences or contigs.fasta or scaffolds.fasta file? Thanks for your help.

**Reply by asl** [reply]
BinSPreader manual (https://cab.spbu.ru/software/binspreader/) explicitly says to bin scaffolds.
